solang-server/src/serverutils/backend.rs

```diff
@@ -87,6 +97,693 @@ impl Backend {
+                for expp in args {
+                    Backend::construct_expr(expp, lookup_tbl, symtab, ns);
+                }
+            }
+            ExternalFunctionCall {
+                loc: _,
+                returns: _,
+                contract_no: _,
+                function_no: _,
+                address,
+                args,
+                value,
+                gas,
+            } => {
+                Backend::construct_expr(address, lookup_tbl, symtab, ns);
+                for expp in args {
+                    Backend::construct_expr(expp, lookup_tbl, symtab, ns);
+                }
+
+                Backend::construct_expr(value, lookup_tbl, symtab, ns);
+                Backend::construct_expr(gas, lookup_tbl, symtab, ns);
+            }
+            ExternalFunctionCallRaw {
+                loc: _,
+                ty: _,
+                address,
+                args,
+                value,
+                gas,
+            } => {
+                Backend::construct_expr(args, lookup_tbl, symtab, ns);
+                Backend::construct_expr(address, lookup_tbl, symtab, ns);
+                Backend::construct_expr(value, lookup_tbl, symtab, ns);
+                Backend::construct_expr(gas, lookup_tbl, symtab, ns);
+            }
+            Constructor {
+                loc: _,
+                contract_no: _,
+                constructor_no: _,
+                args,
+                gas,
+                value,
+                salt,
+            } => {
+                Backend::construct_expr(gas, lookup_tbl, symtab, ns);
+                for expp in args {
+                    Backend::construct_expr(expp, lookup_tbl, symtab, ns);
+                }
+                if let Some(optval) = value {
+                    Backend::construct_expr(optval, lookup_tbl, symtab, ns);
+                }
+                if let Some(optsalt) = salt {
+                    Backend::construct_expr(optsalt, lookup_tbl, symtab, ns);
+                }
+            }
+
+            //Hash table operation expression
+            Keccak256(_locs, _typ, expr) => {
+                for expp in expr {
+                    Backend::construct_expr(expp, lookup_tbl, symtab, ns);
+                }
+                lookup_tbl.push((
+                    _locs.1 as u64,
+                    _locs.2 as u64,
+                    String::from("Keccak256 hash"),
+                ));
+            }
+
+            ReturnData(locs) => {
+                let msg = String::from("Return");
+                lookup_tbl.push((locs.1 as u64, locs.2 as u64, msg));
+            }
+            GetAddress(locs, _typ) => {
+                let msg = String::from("Get address");
+                lookup_tbl.push((locs.1 as u64, locs.2 as u64, msg));
+            }
+            Balance(_locs, _typ, expr) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+            }
+            Builtin(_locs, _typ, _builtin, expr) => {
```

Comment:
The builtins could do with documentation, but solang needs to add those first.

Reply:
will fix in next eval